src/stamps.ts

```diff
@@ -203,6 +229,78 @@ export class StampsManager {
     }
   }
 
+  public async refreshStampsExtends(config: StampsConfigExtends, beeDebug: BeeDebug): Promise<void> {
+    stampCheckCounter.inc()
+    logger.info('checking postage stamps')
+    const stamps = await beeDebug.getAllPostageBatch()
+    logger.debug('retrieved stamps', stamps)
+
+    const { amount, ttlMin, depth } = config
+
+    // Get all usable stamps sorted by usage from most used to least
+    this.usableStamps = filterUsableStampsExtends(stamps)
+
+    if (!this.isBuyingStamp) {
+      if (this.usableStamps.length === 0) {
+        try {
+          this.isBuyingStamp = true
+          const { stamp } = await buyNewStamp(depth, amount, beeDebug)
+
+          // Add the bought postage stamp
+          this.usableStamps.push(stamp)
```

Comment:
Why do you add it here? I don't think it is then used anywhere. Instead, it will be overridden next time when this method runs with the `filterUsableStampsExtends()` call...

Reply:
Also as when you buy the new stamp, that stamp might not be usable right after purchase, so basically you might be circumventing the `filterUsableStampsExtends` filter here.

Reply:
> Why do you add it here? I don't think it is then used anywhere. Instead, it will be overridden next time when this method runs with the `filterUsableStampsExtends()` call...

The reason this is here is in case there is no stamp created, it will create one which will be monitored from that time being. `filterUsableStampsExtends()` will only filter the `usable` stamps currently available. I might be missing your point

Reply:
BTW, once the `buyNewStamp()` a new stamp is created and it won't run anymore

Reply:
> Also as when you buy the new stamp, that stamp might not be usable right after purchase, so basically you might be circumventing the `filterUsableStampsExtends` filter here.

I'm putting the flag `waitForUsable: true`on the `createPostageBatch()` call
https://github.com/ethersphere/gateway-proxy/pull/329/files/81d43459ca773314ced8a850c8c43ef0e4af8f4f#diff-41b27e37dbc88037f052bfae49b6dc06e2cd71248b06ab8537b761900df29e2bR133